tools/vp20compiler/config.h

```diff
@@ -10,17 +10,8 @@
 #define MAX_NV_VERTEX_PROGRAM_TEMPS         13
 #define MAX_NV_VERTEX_PROGRAM_PARAMS       192
 #define MAX_NV_VERTEX_PROGRAM_INPUTS        16
-/*
- Actually Xbox has 11 outputs, however, the parser is using the GL spec and
- expects 8 units. From the GL spec (NV_vertex_program, section 2.14.1.5):
-
-   "If the number of texture units supported is less than eight, the values of
-    vertex result registers that do not correspond to existent texture units
-    are ignored."
-
- So despite only having 4 texture units, we provide 8 texture units here.
- = 11 actual outputs (including 4 texture units) + 4 fake texture-units.
-*/
-#define MAX_NV_VERTEX_PROGRAM_OUTPUTS       (11+4)
+#define MAX_HARDWARE_INPUTS                 16
+#define MAX_NV_VERTEX_PROGRAM_OUTPUTS       15
+#define MAX_HARDWARE_OUTPUTS                15
```

Comment:
nit: This is a bit arbitrarily sized.

The field is actually 8 bits, so theoretically we should have 256 registers. However, register index 0xF is index register `A0` (according to XQEMU code), so it's safe to assume that the registers above are garbage or aliases.

In a future PR I plan to remove these preprocessor defines to replace them with a more dynamic system which could be used to experiment with higher output indices. For now, 15 is fine.